R/mod_downloadHandler.R

```diff
@@ -40,45 +40,101 @@ mod_downloadHandler_filetype_ui <- function(id){
 mod_downloadHandler_include_ui <- function(id){
   # will want to change this to input_UI so we can include additional items
   # such as "Include comments" check boxes for summary, maint-metrics, comm usage, and overall comments
-  my_choices <- c("Report Author", "Report Date", "Risk Score", "Overall Comment", "Package Summary",
-    "Maintenance Metrics", "Maintenance Comments", "Community Usage Metrics", "Community Usage Comments",
-    "Source Explorer Comments")
-  div(
-    strong(p("Elements to include:")),
-    div(align = 'left', class = 'twocol', style = 'margin-top: 0px;',
-      # checkboxGroupInput(
-      shinyWidgets::prettyCheckboxGroup(
-        NS(id, "report_includes"), label = NULL, inline = FALSE,
-        choices = my_choices, selected = my_choices
-      )
-    )
-  )
+  uiOutput(NS(id, "mod_downloadHandler_incl_output"))
 }
-
-#' downloadHandler Inlcude Server Function
+#' downloadHandler Include Server Function
 #'
 #' @description A shiny Module.
 #'
 #' @param id Internal parameters for {shiny}.
+#' @param pkg_name the name of the package passed by mod_reportPreview
+#' @param user char vector containing name and role
+#' @param parent the parent (calling module) session information
+#' 
+#' @importFrom shiny showModal modalDialog
+#' @importFrom glue glue
+#' @importFrom shinyWidgets prettyCheckboxGroup updatePrettyCheckboxGroup
 #'
 #' @noRd 
-mod_downloadHandler_include_server <- function(id) {
+mod_downloadHandler_include_server <- function(id, pkg_name, user, parent) {
   moduleServer(id, function(input, output, session) {
+    ns <- session$ns
+    
+    rpt_choices <- c("Report Author", "Report Date", "Risk Score", "Overall Comment", "Package Summary",
+                     "Maintenance Metrics", "Maintenance Comments", "Community Usage Metrics", "Community Usage Comments",
+                     "Source Explorer Comments")
+    
+    counter <- reactiveVal(value = 0L)
+    user_file <- reactiveVal(value = NULL)
+    
+    output$mod_downloadHandler_incl_output <- renderUI({
+      div(
+        strong(p("Elements to include:")),
+        div(align = 'left', class = 'twocol', style = 'margin-top: 0px;',
+            shinyWidgets::prettyCheckboxGroup(
+              ns("report_includes"), label = NULL, inline = FALSE,
+              choices = rpt_choices, selected = rpt_choices
+            )
+        ),
+        actionButton(ns("store_prefs"), "Store Preferences")
+      )
+    })
+    
     observe({
-      # print("#####################################")
-      # print("input$report_includes:")
-      # print(input$report_includes)
-      # shinyWidgets::updatePrettyCheckboxGroup("report_includes",)
+     req(counter() == 0L)
+     req(parent$input$tabs == "Build Report")
+     
+    # retrieve user data, if it exists.  Otherwise use rpt_choices above.
+     user_file(glue::glue("./inst/report_downloads/report_prefs_{user$name}_{user$role}.txt"))
```

Comment:
I do not think this file location will work as currently constructed for normal use case (i.e. `riskassessment::run_app()`). I tried running the app in a different directory and I got this error upon exiting.
![image](https://github.com/pharmaR/riskassessment/assets/85879620/ad15fca3-8fdd-44c2-9676-a3bbcb180b60)

Reply:
using system.file() to construct the file name now